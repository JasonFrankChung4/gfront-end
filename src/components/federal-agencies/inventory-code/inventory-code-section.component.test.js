import React from 'react'
import { shallow } from 'enzyme'
import InventoryCodeSectionComponent from './inventory-code-section.component'

const props = {
  entry: ['version', { description: 'The version of the metadata schema in use.', type: 'string' }],
  isRequired: true,
  toggleDetails: jest.fn()
}

let wrapper
describe('components - InventoryCodeSectionComponent', () => {
  beforeEach(() => {
    wrapper = shallow(<InventoryCodeSectionComponent {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render proper background color for top level entries', () => {
    expect(wrapper.find('tr').prop('className')).toContain('top-level')

    wrapper.setProps({ indent: 1 })
    expect(wrapper.find('tr').prop('className')).not.toContain('partial')
  })

  it('should render proper background color only for top level entries', () => {
    expect(wrapper.find('tr').prop('className')).toContain('top-level')

    wrapper.setProps({ isRequired: false })
    expect(wrapper.find('tr').prop('className')).toContain('optional')
  })

  it('should render drop down arrow only if it contains sub-entries', () => {
    expect(wrapper.find('.dropdown > div').exists()).toBeFalsy()

    wrapper.setProps({
      entry: [
        'measurementType',
        {
          properties: {
            ifOther: { type: 'string', dsecription: 'A one- or two- sentence description...' }
          },
          type: 'object'
        }
      ]
    })
    expect(wrapper.find('.dropdown > div').prop('className')).toContain('arrow-up-or-down')
  })

  it('should hide drop down on click', () => {
    expect(wrapper.state('dropDown')).toBeTruthy()

    wrapper.setProps({
      entry: [
        'measurementType',
        {
          properties: {
            ifOther: { type: 'string', dsecription: 'A one- or two- sentence description...' }
          },
          type: 'object'
        }
      ]
    })

    console.log(wrapper.debug())

    wrapper.find('.dropdown').simulate('click')
    expect(wrapper.state('dropDown')).toBeFalsy()
  })
})
