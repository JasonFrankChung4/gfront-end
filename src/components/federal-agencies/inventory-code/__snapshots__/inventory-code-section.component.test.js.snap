// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - InventoryCodeSectionComponent should render correctly 1`] = `
<Fragment>
  <tr
    className="top-level"
  >
    <th
      className="field-name-text"
      scope="row"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        className="dropdown"
      />
      version
      <button
        className="details"
        onClick={[Function]}
      >
        <div
          className="details-text"
        >
          details
        </div>
        <div
          className="details-arrow arrow-right"
        />
      </button>
    </th>
    <td
      className="data-type"
    >
      string
    </td>
    <td
      className="description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "The version of the metadata schema in use.",
        }
      }
    />
  </tr>
</Fragment>
`;
